packages/openapi/src/openapi-request-builder.ts

```diff
@@ -160,10 +160,11 @@ export class OpenApiRequestBuilder<ResponseT = any> {
   }
 
   private getHeaders(): OriginOptions {
-    if (Object.keys(this.customHeaders).length) {
-      return { custom: this.customHeaders, requestConfig: {} };
-    }
-    return { requestConfig: {} };
+    // if (Object.keys(this.customHeaders).length) {
+    //   return { custom: this.customHeaders, requestConfig: {} };
+    // }
+    // return { requestConfig: {} };
+    return { requestConfig: this.parameters?.headerParameters || {} };
```

Comment:
[req]
1. Please remove the old comment
2. I think this is incorrect because you want to send along custom headers if there are any. Please make sure this works for both cases.

```suggestion
const options = { requestConfig: this.parameters?.headerParameters || {} };
if (Object.keys(this.customHeaders).length) {
  return { custom: this.customHeaders, ...options };
}
return options;
```